great_expectations/checkpoint/actions.py

```diff
@@ -48,6 +48,14 @@ class ValidationAction:
     def __init__(self, data_context) -> None:
         self.data_context = data_context
 
+    @property
+    def _cloud_mode(self) -> bool:
+        from great_expectations.data_context.data_context.cloud_data_context import (
+            CloudDataContext,
+        )
+
+        return isinstance(self.data_context, CloudDataContext)
```

Comment:
It's still not clear to me why it's not better to have the base DataContext class have a `data_context.cloud_mode` attribute that defaults to false.
I don't think the extra code churn here actually makes the abstraction any better. But maybe I just don't understand it.

It should go without saying that we should work to abstract away these conditional `if cloud_mode` checks.

Reply:
These checks definitely need to go but as it stands, there is quite a bit of Cloud leakage into other parts of the codebase. Since this property is simply encapsulating that leakage here, I'd vote to keep this for now and then work to decouple Cloud in a subsequent effort.

Just to clarify, we're not adding a new behavior here - instead of checking for `cloud_mode`, we're evaluating whether or not we're in a Cloud backed environment through the context type (different ways to achieve the same goal).

Happy to brainstorm a better way to do this.

Reply:
@cdkini I understand that we aren't adding new behavior here. I also accept that this sort of check is still needed.

What I'm saying is that I don't think only having the `cloud_mode` attribute on the `CloudDataContext` really improves things, since you still have to check for it on every `DataContext` anyways. Except now it's harder because we are wary of `AttributeErrors`.

Reply:
oh no we removed `cloud_mode` as an attr now - hence the change to `isinstance` as a check

Reply:
>oh no we removed cloud_mode as an attr now - hence the change to isinstance as a check

I understand that `cloud_mode` is only an attr on  `CloudDataContext`.